server/auth/types/jwt/jwt_auth.ts

```diff
@@ -13,136 +13,116 @@
  *   permissions and limitations under the License.
  */
 
-import { parse, format } from 'url';
-import { get } from 'lodash';
 import { ParsedUrlQuery } from 'querystring';
-import { SecurityPluginConfigType } from '../../..';
 import {
   SessionStorageFactory,
   IRouter,
   ILegacyClusterClient,
   CoreSetup,
-  AuthenticationHandler,
   KibanaRequest,
   Logger,
-} from '../../../../../../src/core/server';
+  LifecycleResponseFactory,
+  AuthToolkit,
+  IKibanaResponse,
+} from 'kibana/server';
+import { SecurityPluginConfigType } from '../../..';
 import { SecuritySessionCookie } from '../../../session/security_cookie';
-import { SecurityClient } from '../../../backend/opendistro_security_client';
-import { User } from '../../user';
-import { IAuthenticationType } from '../authentication_type';
+import { AuthenticationType } from '../authentication_type';
 
-export class JwtAuthentication implements IAuthenticationType {
+export class JwtAuthentication extends AuthenticationType {
   public readonly type: string = 'jwt';
 
   private authHeaderName: string;
-  private securityClient: SecurityClient;
 
   constructor(
-    private readonly config: SecurityPluginConfigType,
-    private readonly sessionStorageFactory: SessionStorageFactory<SecuritySessionCookie>,
-    private readonly router: IRouter,
-    private readonly esClient: ILegacyClusterClient,
-    private readonly coreSetup: CoreSetup,
-    private readonly logger: Logger
+    config: SecurityPluginConfigType,
+    sessionStorageFactory: SessionStorageFactory<SecuritySessionCookie>,
+    router: IRouter,
+    esClient: ILegacyClusterClient,
+    coreSetup: CoreSetup,
+    logger: Logger
   ) {
+    super(config, sessionStorageFactory, router, esClient, coreSetup, logger);
     this.authHeaderName = this.config.jwt?.header.toLowerCase() || 'authorization';
-    this.securityClient = new SecurityClient(esClient);
   }
 
-  authHandler: AuthenticationHandler = async (request, response, tookit) => {
-    const jwtToken = this.getJwtToken(request);
-    if (jwtToken) {
-      const authHeaderValue = `Bearer ${jwtToken}`;
-      let user: User;
-      try {
-        user = await this.securityClient.authenticateWithHeader(
-          request,
-          this.authHeaderName,
-          authHeaderValue
-        );
-      } catch (error) {
-        console.log(error);
-        return response.unauthorized();
+  private getBearerToken(request: KibanaRequest): string | undefined {
```

Comment:
Looks like we can extract a private function

getToken(){
 const urlParamName = this.config.jwt?.url_param;
    if (urlParamName) {
      return token = (request.url.query as ParsedUrlQuery)[urlParamName];
};

Then getBearerToken, requestIncludesAuthInfo and getAdditionalAuthHeader can all it to reuse some common logic.

Reply:
updated accordingly